### PR TITLE
Fix excessive setParam calls

### PR DESCRIPTION
**Pull request by chitang**

### CODE REVIEW

File: version.sbt [hunk 1]
```diff
@@ -1 +1 @@
-ThisBuild / version := "4.0.31-SNAPSHOT"
+ThisBuild / version := "4.1.0-SNAPSHOT"
```
Comment (sjzavaleta) [inline version.sbt:1]: is this related?

Comment (chitang) [reply version.sbt:1]: No, but I missed the opportunity to ask for a bigger version change lately.  We don't really strictly follow semantic versioning.  There are changes that are significant enough to have a bigger version bump.

Comment (chitang) [reply version.sbt:1]: We should have a major bump everytime we change analysis (de)serialization.

**Review by sjzavaleta — APPROVED** [review]
